app/src/main/java/com/picone/go4lunch/presentation/ui/MapsFragment.java

```diff
@@ -23,12 +50,85 @@ public static MapsFragment newInstance() {
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mFusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(requireActivity());
+        fetchLastLocation();
     }
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         mBinding = FragmentMapsBinding.inflate(inflater, container, false);
+        mBinding.mapView.onCreate(savedInstanceState);
+        mBinding.mapView.onResume();
+        try {
+            MapsInitializer.initialize(requireActivity().getApplicationContext());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
         return mBinding.getRoot();
     }
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        updateLocationUI();
+    }
+
+
+    private void fetchLastLocation() {
+        if (ActivityCompat.checkSelfPermission(this.requireContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this.requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(requireActivity(), new String[]
+                    {ACCESS_FINE_LOCATION}, REQUEST_CODE);
+            return;
+        }
+
+        Task<Location> task = mFusedLocationProviderClient.getLastLocation();
+        task.addOnSuccessListener(location -> {
+            if (location != null) {
+                mCurrentLocation = location;
+
+                mBinding.mapView.getMapAsync(this);
+            }
+        });
+    }
+
+    private void updateLocationUI() {
+        if (mMap == null) {
+            return;
+        }
+        try {
+            if (mLocationPermissionGranted) {
+                mMap.setMyLocationEnabled(true);
+                mMap.getUiSettings().setMyLocationButtonEnabled(true);
+                setUpMapCurrentPosition();
+            } else {
+                mMap.setMyLocationEnabled(false);
+                mMap.getUiSettings().setMyLocationButtonEnabled(false);
+                getLocationPermission();
+            }
+        } catch (SecurityException e) {
```

Comment:
Notify User instead of just logging

Reply:
Done with toast. Do i have to put a loading bar to notify that login is working?

Reply:
Yes display a loader everytime your app is performing long time tasks. Use Lottie. Easy to set up.

Reply:
Ok i'll work around Lottie when i'll fixed navigation with auth